sky/shell/platform/mac/platform_service_provider.cc

```diff
@@ -19,33 +19,43 @@ void PlatformServiceProvider::ConnectToService(
   if (service_name == mojo::NetworkService::Name_) {
     network_.Create(
         nullptr, mojo::MakeRequest<mojo::NetworkService>(client_handle.Pass()));
+    return;
   }
 #if TARGET_OS_IPHONE
   if (service_name == ::keyboard::KeyboardService::Name_) {
     keyboard_.Create(nullptr, mojo::MakeRequest<::keyboard::KeyboardService>(
                                   client_handle.Pass()));
+    return;
   }
   if (service_name == ::media::MediaPlayer::Name_) {
     media_player_.Create(
         nullptr, mojo::MakeRequest<::media::MediaPlayer>(client_handle.Pass()));
+    return;
   }
   if (service_name == ::media::MediaService::Name_) {
     media_service_.Create(nullptr, mojo::MakeRequest<::media::MediaService>(
                                        client_handle.Pass()));
+    return;
   }
   if (service_name == ::vsync::VSyncProvider::Name_) {
     vsync_.Create(nullptr, mojo::MakeRequest<::vsync::VSyncProvider>(
                                client_handle.Pass()));
+    return;
   }
   if (service_name == ::activity::PathService::Name_) {
     path_.Create(nullptr, mojo::MakeRequest<::activity::PathService>(
                               client_handle.Pass()));
+    return;
   }
   if (service_name == ::activity::Activity::Name_) {
     activity_.Create(
         nullptr, mojo::MakeRequest<::activity::Activity>(client_handle.Pass()));
+    return;
   }
 #endif
+
+  DCHECK(false) << "The platform service provider cannot find a service for '"
+                << service_name.data() << "'";
```

Comment:
I don't think this DCHECK is correct.  The proper way to signal this condition is to close the pipe.